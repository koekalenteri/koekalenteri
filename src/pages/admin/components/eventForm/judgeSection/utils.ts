import type { DeepPartial, EventClass, EventType, Judge, PublicJudge } from '../../../../../types'
import type { PartialEvent } from '../../EventForm'

import { isSameDay } from 'date-fns'

export type PartialPublicJudge = Partial<PublicJudge>
export type PartialPublicJudgeValue = PartialPublicJudge | PartialPublicJudge[]

export const filterJudges = (
  judges: Judge[],
  eventJudges: PublicJudge[],
  id: number | undefined,
  eventType?: EventType
) =>
  judges
    .filter((j) => !eventType?.official || j.eventTypes.includes(eventType.eventType))
    .filter((j) => j.id === id || !eventJudges.find((ej) => ej.id === j.id))

export const hasJudge = (c: DeepPartial<EventClass>, id?: number) =>
  Array.isArray(c.judge) ? c.judge.find((j) => j.id === id) : c.judge?.id === id

export const filterClassesByJudgeId = (classes?: EventClass[], id?: number) => classes?.filter((c) => hasJudge(c, id))

export const toArray = (j?: PartialPublicJudge): PartialPublicJudge[] => (j ? [j] : [])
export const makeArray = (j?: PartialPublicJudgeValue) => (Array.isArray(j) ? [...j] : toArray(j))
export const selectJudge = (j?: PartialPublicJudgeValue, judge?: PublicJudge): PartialPublicJudge[] => {
  const a = makeArray(j)
  if (judge) {
    const index = a.findIndex((cj) => cj.id === judge.id)
    if (index === -1) {
      a.push(judge)
    } else {
      a[index] = judge
    }
  }
  return a
}

export const removeJudge = (j?: PartialPublicJudgeValue, id?: number): PartialPublicJudge[] => {
  const a = makeArray(j)
  return a.filter((cj) => cj.id !== id)
}

export const updateJudge = (
  event: PartialEvent,
  id: number | undefined,
  judge: PublicJudge | undefined,
  values?: DeepPartial<EventClass>[]
) => {
  const isSelected = (c: DeepPartial<EventClass>) =>
    values?.find((v) => event && isSameDay(v.date ?? event.startDate, c.date ?? event.startDate) && v.class === c.class)
  return event.classes?.map((c) => ({
    ...c,
    judge: isSelected(c) ? selectJudge(c.judge, judge) : removeJudge(c.judge, id),
  }))
}
