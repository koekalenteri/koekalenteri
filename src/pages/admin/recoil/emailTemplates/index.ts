export * from './actions'
export * from './atoms'
export * from './atoms2'
export * from './effects'
export * from './selectors'
