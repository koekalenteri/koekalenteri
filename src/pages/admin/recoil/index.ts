export * from './emailTemplates'
export * from './events'
export * from './eventTypes'
export * from './officials'
export * from './organizers'
export * from './registrations'
export * from './user'
