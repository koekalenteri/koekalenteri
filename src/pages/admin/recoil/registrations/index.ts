export * from './atoms'
// 2nd file to avoid circular deps
export * from './atoms2'
export * from './selectors'
