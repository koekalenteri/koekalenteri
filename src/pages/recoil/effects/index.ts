export * from './log'
export * from './storage'

export const getParamFromFamilyKey = (key: string) => key.split('__')[1]?.slice(1, -1)
