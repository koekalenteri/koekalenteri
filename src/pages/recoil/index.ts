export * from './effects'
export * from './events'
export * from './judges'
export * from './registration'
export * from './user'
