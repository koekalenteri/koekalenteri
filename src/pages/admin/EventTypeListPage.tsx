import { useCallback, useState } from 'react'
import { useTranslation } from 'react-i18next'
import AddCircleOutline from '@mui/icons-material/AddCircleOutline'
import CloudSync from '@mui/icons-material/CloudSync'
import Stack from '@mui/material/Stack'
import { useRecoilState, useRecoilValue } from 'recoil'

import StyledDataGrid from '../components/StyledDataGrid'
import { isAdminSelector } from '../recoil'

import FullPageFlex from './components/FullPageFlex'
import { QuickSearchToolbar } from './components/QuickSearchToolbar'
import AutoButton from './eventListPage/AutoButton'
import { useEventTypeListPageColumns } from './eventTypeListPage/columns'
import { CreateEventTypeDialog } from './eventTypeListPage/CreateEventTypeDialog'
import { eventTypeFilterAtom, filteredEventTypesSelector, useAdminEventTypeActions } from './recoil'

export default function EventTypeListPage() {
  const { t } = useTranslation()
  const eventTypes = useRecoilValue(filteredEventTypesSelector)
  const isAdmin = useRecoilValue(isAdminSelector)
  const actions = useAdminEventTypeActions()
  const [searchText, setSearchText] = useRecoilState(eventTypeFilterAtom)
  const [createOpen, setCreateOpen] = useState(false)

  const columns = useEventTypeListPageColumns()

  const onChange = useCallback(
    (event: React.ChangeEvent<HTMLInputElement>) => setSearchText(event.target.value),
    [setSearchText]
  )

  const clearSearch = useCallback(() => setSearchText(''), [setSearchText])
  const createAction = useCallback(() => setCreateOpen(true), [])

  return (
    <FullPageFlex>
      <Stack direction="row" spacing={2}>
        <AutoButton
          disabled={!isAdmin}
          startIcon={<AddCircleOutline />}
          onClick={createAction}
          text={t('eventType.create')}
        />
        <AutoButton
          startIcon={<CloudSync />}
          onClick={actions.refresh}
          sx={{ display: isAdmin ? undefined : 'none' }}
          text={t('updateData', { data: 'eventTypes' })}
        />
      </Stack>

      <StyledDataGrid
        columns={columns}
        slots={{ toolbar: QuickSearchToolbar }}
        slotProps={{
          toolbar: {
            value: searchText,
            onChange,
            clearSearch,
          },
        }}
        rows={eventTypes}
        getRowId={(row) => row.eventType}
      />
      <CreateEventTypeDialog onClose={() => setCreateOpen(false)} open={createOpen} />
    </FullPageFlex>
  )
}
