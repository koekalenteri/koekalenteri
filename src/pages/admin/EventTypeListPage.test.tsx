import { Suspense } from 'react'
import { MemoryRouter } from 'react-router-dom'
import { ThemeProvider } from '@mui/material'
import { render } from '@testing-library/react'
import { SnackbarProvider } from 'notistack'
import { RecoilRoot } from 'recoil'

import theme from '../../assets/Theme'
import { flushPromises } from '../../test-utils/utils'

import EventTypeListPage from './EventTypeListPage'

jest.mock('../../api/eventType')
jest.mock('../../api/user')

describe('EventTypeListPage', () => {
  beforeAll(() => jest.useFakeTimers())
  afterEach(() => jest.runOnlyPendingTimers())
  afterAll(() => jest.useRealTimers())

  it('renders', async () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <RecoilRoot>
          <MemoryRouter>
            <Suspense fallback={<div>loading...</div>}>
              <SnackbarProvider>
                <EventTypeListPage />
              </SnackbarProvider>
            </Suspense>
          </MemoryRouter>
        </RecoilRoot>
      </ThemeProvider>
    )
    await flushPromises()
    expect(container).toMatchSnapshot()
  })
})
