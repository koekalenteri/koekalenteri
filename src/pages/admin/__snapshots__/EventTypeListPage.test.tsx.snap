// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventTypeListPage renders 1`] = `
<div>
  <div
    class="MuiBox-root css-1uovp08"
  >
    <div
      class="MuiBox-root css-10784zt"
    >
      <div
        class="MuiStack-root css-1hezniw-MuiStack-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1qsu95a-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1sh91j5-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-20bmp1-MuiSvgIcon-root"
              data-testid="AddCircleOutlineIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8"
              />
            </svg>
          </span>
          eventType.create
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1qsu95a-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1sh91j5-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-20bmp1-MuiSvgIcon-root"
              data-testid="CloudSyncIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M21.5 14.98c-.02 0-.03 0-.05.01C21.2 13.3 19.76 12 18 12c-1.4 0-2.6.83-3.16 2.02C13.26 14.1 12 15.4 12 17c0 1.66 1.34 3 3 3l6.5-.02c1.38 0 2.5-1.12 2.5-2.5s-1.12-2.5-2.5-2.5M10 4.26v2.09C7.67 7.18 6 9.39 6 12c0 1.77.78 3.34 2 4.44V14h2v6H4v-2h2.73C5.06 16.54 4 14.4 4 12c0-3.73 2.55-6.85 6-7.74M20 6h-2.73c1.43 1.26 2.41 3.01 2.66 5h-2.02c-.23-1.36-.93-2.55-1.91-3.44V10h-2V4h6z"
              />
            </svg>
          </span>
          updateData data
        </button>
      </div>
      <div
        class="MuiDataGrid-root MuiDataGrid-root--densityCompact MuiDataGrid-withBorderColor css-slqvgp-MuiDataGrid-root"
        style="--DataGrid-width: 0px; --DataGrid-hasScrollX: 1; --DataGrid-hasScrollY: 1; --DataGrid-scrollbarSize: 0px; --DataGrid-rowWidth: 340px; --DataGrid-columnsTotalWidth: 340px; --DataGrid-leftPinnedWidth: 0px; --DataGrid-rightPinnedWidth: 0px; --DataGrid-headerHeight: 39px; --DataGrid-headersTotalHeight: 39px; --DataGrid-topContainerHeight: 39px; --DataGrid-bottomContainerHeight: 0px; --height: 28px;"
      >
        <div
          class="MuiStack-root css-k5y4k9-MuiStack-root"
        >
          <div
            class="MuiDataGrid-toolbarContainer css-1wrr6qg-MuiDataGrid-toolbarContainer"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1pupl87-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-11t8gxv-MuiInputBase-root-MuiInput-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-18w7uxr-MuiSvgIcon-root"
                  data-testid="SearchIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
                  />
                </svg>
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-1o3an4k-MuiInputBase-input-MuiInput-input"
                  id=":r1:"
                  placeholder="search.placeholder"
                  type="text"
                  value=""
                />
                <button
                  aria-label="clear"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-hizwh9-MuiButtonBase-root-MuiIconButton-root"
                  style="visibility: hidden;"
                  tabindex="0"
                  title="clear"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-18w7uxr-MuiSvgIcon-root"
                    data-testid="ClearIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-colcount="4"
          aria-multiselectable="false"
          aria-rowcount="2"
          class="MuiDataGrid-main css-3eek4p-MuiDataGrid-main"
          role="grid"
          tabindex="-1"
        >
          <div
            class="css-qtdhae"
            data-id="gridPanelAnchor"
            role="presentation"
          />
          <div
            class="MuiDataGrid-virtualScroller MuiDataGrid-virtualScroller--hasScrollX css-1793420-MuiDataGrid-virtualScroller"
            role="presentation"
            style="overflow-x: hidden;"
          >
            <div
              class="MuiDataGrid-topContainer MuiDataGrid-container--top css-2sp4oa"
              role="presentation"
            >
              <div
                class="MuiDataGrid-columnHeaders css-1pmk00y-MuiDataGrid-columnHeaders"
                role="presentation"
              >
                <div
                  aria-rowindex="1"
                  class="MuiDataGrid-row--borderBottom css-yrdy0g-MuiDataGrid-columnHeaderRow"
                  role="row"
                >
                  <div
                    role="presentation"
                    style="width: 0px;"
                  />
                  <div
                    aria-colindex="1"
                    aria-label="eventType.eventType"
                    aria-sort="none"
                    class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable MuiDataGrid-withBorderColor"
                    data-field="eventType"
                    role="columnheader"
                    style="height: 39px; width: 180px;"
                    tabindex="0"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderDraggableContainer"
                      draggable="false"
                      role="presentation"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainer"
                        role="presentation"
                      >
                        <div
                          class="MuiDataGrid-columnHeaderTitleContainerContent"
                        >
                          <div
                            aria-label=""
                            class="MuiDataGrid-columnHeaderTitle css-1qb993p-MuiDataGrid-columnHeaderTitle"
                            data-mui-internal-clone-element="true"
                          >
                            eventType.eventType
                          </div>
                        </div>
                        <div
                          class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                        >
                          <button
                            aria-label="Sort"
                            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-hizwh9-MuiButtonBase-root-MuiIconButton-root"
                            field="eventType"
                            tabindex="-1"
                            title="Sort"
                            type="button"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-18w7uxr-MuiSvgIcon-root"
                              data-testid="ArrowUpwardIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                              />
                            </svg>
                          </button>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--resizable MuiDataGrid-columnSeparator--sideRight"
                      style="min-height: 39px;"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-20bmp1-MuiSvgIcon-root"
                        data-testid="SeparatorIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <rect
                          height="24"
                          rx="0.5"
                          width="1"
                          x="11.5"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    aria-colindex="2"
                    aria-label="official"
                    aria-sort="none"
                    class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable MuiDataGrid-withBorderColor"
                    data-field="official"
                    role="columnheader"
                    style="height: 39px; width: 80px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderDraggableContainer"
                      draggable="false"
                      role="presentation"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainer"
                        role="presentation"
                      >
                        <div
                          class="MuiDataGrid-columnHeaderTitleContainerContent"
                        >
                          <div
                            aria-label=""
                            class="MuiDataGrid-columnHeaderTitle css-1qb993p-MuiDataGrid-columnHeaderTitle"
                            data-mui-internal-clone-element="true"
                          >
                            official
                          </div>
                        </div>
                        <div
                          class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                        >
                          <button
                            aria-label="Sort"
                            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-hizwh9-MuiButtonBase-root-MuiIconButton-root"
                            field="official"
                            tabindex="-1"
                            title="Sort"
                            type="button"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-18w7uxr-MuiSvgIcon-root"
                              data-testid="ArrowUpwardIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                              />
                            </svg>
                          </button>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--resizable MuiDataGrid-columnSeparator--sideRight"
                      style="min-height: 39px;"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-20bmp1-MuiSvgIcon-root"
                        data-testid="SeparatorIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <rect
                          height="24"
                          rx="0.5"
                          width="1"
                          x="11.5"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    aria-colindex="3"
                    aria-label="active"
                    aria-sort="none"
                    class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable MuiDataGrid-withBorderColor"
                    data-field="active"
                    role="columnheader"
                    style="height: 39px; width: 80px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderDraggableContainer"
                      draggable="false"
                      role="presentation"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainer"
                        role="presentation"
                      >
                        <div
                          class="MuiDataGrid-columnHeaderTitleContainerContent"
                        >
                          <div
                            aria-label=""
                            class="MuiDataGrid-columnHeaderTitle css-1qb993p-MuiDataGrid-columnHeaderTitle"
                            data-mui-internal-clone-element="true"
                          >
                            active
                          </div>
                        </div>
                        <div
                          class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                        >
                          <button
                            aria-label="Sort"
                            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-hizwh9-MuiButtonBase-root-MuiIconButton-root"
                            field="active"
                            tabindex="-1"
                            title="Sort"
                            type="button"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-18w7uxr-MuiSvgIcon-root"
                              data-testid="ArrowUpwardIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                              />
                            </svg>
                          </button>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--resizable MuiDataGrid-columnSeparator--sideRight"
                      style="min-height: 39px;"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-20bmp1-MuiSvgIcon-root"
                        data-testid="SeparatorIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <rect
                          height="24"
                          rx="0.5"
                          width="1"
                          x="11.5"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    aria-colindex="4"
                    aria-label="eventType.description"
                    aria-sort="none"
                    class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable MuiDataGrid-withBorderColor MuiDataGrid-columnHeader--last"
                    data-field="description"
                    role="columnheader"
                    style="height: 39px; width: 0px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderDraggableContainer"
                      draggable="false"
                      role="presentation"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainer"
                        role="presentation"
                      >
                        <div
                          class="MuiDataGrid-columnHeaderTitleContainerContent"
                        >
                          <div
                            aria-label=""
                            class="MuiDataGrid-columnHeaderTitle css-1qb993p-MuiDataGrid-columnHeaderTitle"
                            data-mui-internal-clone-element="true"
                          >
                            eventType.description
                          </div>
                        </div>
                        <div
                          class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                        >
                          <button
                            aria-label="Sort"
                            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-hizwh9-MuiButtonBase-root-MuiIconButton-root"
                            field="description"
                            tabindex="-1"
                            title="Sort"
                            type="button"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-18w7uxr-MuiSvgIcon-root"
                              data-testid="ArrowUpwardIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                              />
                            </svg>
                          </button>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--resizable MuiDataGrid-columnSeparator--sideRight"
                      style="min-height: 39px;"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-20bmp1-MuiSvgIcon-root"
                        data-testid="SeparatorIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <rect
                          height="24"
                          rx="0.5"
                          width="1"
                          x="11.5"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-filler"
                    role="presentation"
                  />
                  <div
                    class="MuiDataGrid-scrollbarFiller MuiDataGrid-scrollbarFiller--header"
                    role="presentation"
                  />
                </div>
              </div>
            </div>
            <div
              class="MuiDataGrid-virtualScrollerContent css-1kwdphh-MuiDataGrid-virtualScrollerContent"
              role="presentation"
              style="width: auto; height: 28px;"
            >
              <div
                class="MuiDataGrid-virtualScrollerRenderZone css-s1v7zr-MuiDataGrid-virtualScrollerRenderZone"
                role="rowgroup"
                style="transform: translate3d(0, 0px, 0);"
              >
                <div
                  aria-rowindex="2"
                  aria-selected="false"
                  class="MuiDataGrid-row MuiDataGrid-row--firstVisible MuiDataGrid-row--lastVisible"
                  data-id="TEST1"
                  data-rowindex="0"
                  role="row"
                  style="max-height: 28px; min-height: 28px; --height: 28px;"
                >
                  <div
                    class="MuiDataGrid-cellOffsetLeft"
                    role="presentation"
                    style="width: 0px;"
                  />
                  <div
                    aria-colindex="1"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="0"
                    data-field="eventType"
                    role="gridcell"
                    style="--width: 180px;"
                    tabindex="-1"
                    title="TEST1"
                  >
                    TEST1
                  </div>
                  <div
                    aria-colindex="2"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textCenter"
                    data-colindex="1"
                    data-field="official"
                    role="gridcell"
                    style="--width: 80px;"
                    tabindex="-1"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-x7d0y6-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankOutlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5zm0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2"
                      />
                    </svg>
                  </div>
                  <div
                    aria-colindex="3"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="2"
                    data-field="active"
                    role="gridcell"
                    style="--width: 80px;"
                    tabindex="-1"
                  >
                    <span
                      class="MuiSwitch-root MuiSwitch-sizeSmall css-g5ae86-MuiSwitch-root"
                    >
                      <span
                        class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSuccess PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorSuccess css-11xzqlw-MuiButtonBase-root-MuiSwitch-switchBase"
                      >
                        <input
                          class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                          type="checkbox"
                        />
                        <span
                          class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                        />
                      </span>
                      <span
                        class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                      />
                    </span>
                  </div>
                  <div
                    aria-colindex="4"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="3"
                    data-field="description"
                    role="gridcell"
                    style="--width: 0px;"
                    tabindex="-1"
                    title="TEST1 tapahtymatyyppi"
                  >
                    TEST1 tapahtymatyyppi
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiDataGrid-bottomContainer MuiDataGrid-container--bottom css-14egwce"
              role="presentation"
            />
          </div>
          <div
            aria-hidden="true"
            class="MuiDataGrid-scrollbar MuiDataGrid-scrollbar--vertical css-1b9e9gy"
            tabindex="-1"
          >
            <div
              class="MuiDataGrid-scrollbarContent css-1baulvz"
            />
          </div>
          <div
            aria-hidden="true"
            class="MuiDataGrid-scrollbar MuiDataGrid-scrollbar--horizontal css-1rtad1"
            tabindex="-1"
          >
            <div
              class="MuiDataGrid-scrollbarContent css-1baulvz"
            />
          </div>
        </div>
        <div
          class="MuiDataGrid-footerContainer MuiDataGrid-withBorderColor css-wop1k0-MuiDataGrid-footerContainer"
        >
          <div />
          <div
            class="MuiTablePagination-root css-16mfp94-MuiTablePagination-root"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-16og44m-MuiToolbar-root-MuiTablePagination-toolbar"
            >
              <div
                class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
              />
              <p
                class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
              >
                1–1 of 1
              </p>
              <div
                class="MuiTablePagination-actions"
              >
                <button
                  aria-label="Go to previous page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-1kcqes5-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  title="Go to previous page"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-20bmp1-MuiSvgIcon-root"
                    data-testid="KeyboardArrowLeftIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="Go to next page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-1kcqes5-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  title="Go to next page"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-20bmp1-MuiSvgIcon-root"
                    data-testid="KeyboardArrowRightIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
